File: app/models/cqm/clause_result.rb
module CQM
  # ClauseResult is used to store clause result information in IndividualResult
  class ClauseResult
    include Mongoid::Document
    include Mongoid::Timestamps

    # Library the clause this result is for is in
    field :library_name, type: String
    # Statment the clause this result is for is in
    field :statement_name, type: String
    # LocalId of the clause this result is for
    field :localId, type: String
    # Final, processed result of raw calculation
    field :final, type: String
    # Raw result of clause calculation
    field :raw

    # Relations to other model classes
    embedded_in :individual_result
  end
end

File: app/assets/javascripts/cqm/ClauseResult.js
const mongoose = require('mongoose/browser');

const [String, Mixed] = [
  mongoose.Schema.Types.String,
  mongoose.Schema.Types.Mixed,
];

const ClauseResultSchema = mongoose.Schema({
  // Library the clause this result is for is in
  library_name: String,
  // Statement the clause this result is for is in
  statement_name: String,
  // LocalId of the clause this result is for
  localId: String,
  // Final, processed result of raw calculation
  final: String,
  // Raw result of clause calculation
  raw: Mixed,
});


module.exports.ClauseResultSchema = ClauseResultSchema;
class ClauseResult extends mongoose.Document {
  constructor(object) {
    super(object, ClauseResultSchema);
  }
}
module.exports.ClauseResult = ClauseResult;
